### PR TITLE
feat: add access tags to resources created by the module

### PR DESCRIPTION
**Pull request by MatthewLemmond**
### Description
Adds access tags to the resource instances created by the module if they are passed

#### No release required

- [ ] Examples or tests (addition or updates of examples or tests)
- [ ] Documentation update
- [ ] CI-related update (pipeline, etc.)
- [ ] Other changes that don't affect Terraform code

#### Release required

- [ ] Bug fix (patch release (`x.x.X`): Change that fixes an issue and is compatible with earlier versions)
- [x] New feature (minor release (`x.X.x`): Change that adds functionality and is compatible with earlier versions)
- [ ] Breaking change (major release (`X.x.x`): Change that is likely incompatible with previous versions)

##### Release notes content
Add ability to attach access tags to resources created by the module, consumers who want to use the access tags will need additional permissions for Tagging service at the Administrator level.

---

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### Merge actions for mergers

- Merge by using "Squash and merge".
- Use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents and any release notes provided by the PR author.

    The commit message determines whether a new version of the module is needed, and if so, which semver increment to use (major, minor, or patch).

### CODE REVIEW

**Review by shemau — DISMISSED** [review]
Looks good.

**Review by ocofaigh — CHANGES_REQUESTED** [review]
what about the root level module. Wont it need different variables to pass down to the submodules for access tags?

**Comment by MatthewLemmond** [comment]
Updated to take 3 access tag inputs for each of the different possible resources that could need the access management tags in the same way the normal tags are structured

File: variables.tf [hunk 1]
```diff
@@ -221,6 +247,19 @@ variable "activity_tracker_tags" {
   default     = []
 }
 
+variable "activity_tracker_access_tags" {
+  type        = list(string)
+  description = "A list of access tags to apply to the resources created by the module. For more information, see https://cloud.ibm.com/docs/account?topic=account-access-tags-tutorial."
```
Comment (ocofaigh) [inline variables.tf:252]: `to the resources created by the module` -> `to the Activity Tracker instance created by the module`

File: variables.tf [hunk 2]
```diff
@@ -161,6 +174,19 @@ variable "sysdig_tags" {
   default     = []
 }
 
+variable "sysdig_access_tags" {
+  type        = list(string)
+  description = "A list of access tags to apply to the resources created by the module. For more information, see https://cloud.ibm.com/docs/account?topic=account-access-tags-tutorial."
```
Comment (ocofaigh) [inline variables.tf:179]: `to the resources created by the module` -> `to the Sysdig instance created by the module`

File: variables.tf [hunk 3]
```diff
@@ -83,6 +83,19 @@ variable "logdna_tags" {
   default     = []
 }
 
+variable "logdna_access_tags" {
+  type        = list(string)
+  description = "A list of access tags to apply to the resources created by the module. For more information, see https://cloud.ibm.com/docs/account?topic=account-access-tags-tutorial."
```
Comment (ocofaigh) [inline variables.tf:88]: `to the resources created by the module` -> `to the LogDNA instance created by the module`

**Review by ocofaigh — CHANGES_REQUESTED** [review]
see comments on updates to variable descriptions

**Review by ocofaigh — COMMENTED** [review]
Sorry @MatthewLemmond I just spotted something that you might address in this PR too please. See https://github.com/terraform-ibm-modules/terraform-ibm-observability-instances/pull/282/files#r1200535256

**Comment by ocofaigh** [comment]
Must be a bug somewhere? `Error: ResourceServiceInstanceDoesnotExist: Given service instance : "null" doesn't exist`

**Comment by MatthewLemmond** [comment]
I found it, I capitalized `CRN` when moving to the LoadMapFromYaml function, should be good now.

**Review by ocofaigh — APPROVED** [review]

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**